src/earn/prepareTransactions.test.ts

```diff
@@ -146,6 +147,84 @@ describe('prepareTransactions', () => {
         feeCurrencies: [mockFeeCurrency],
         spendToken: mockToken,
         spendTokenAmount: new BigNumber(5),
+        isGasSubsidized: false,
+      })
+    })
+    it('prepares fees from the cloud function for approve and supply when subsidizing gas fees', async () => {
+      mockFetch.mockResponseOnce(
+        JSON.stringify({
+          status: 'OK',
+          simulatedTransactions: [
+            {
+              status: 'success',
+              blockNumber: '1',
+              gasNeeded: 3000,
+              gasUsed: 2800,
+              gasPrice: '1',
+            },
+            {
+              status: 'success',
+              blockNumber: '1',
+              gasNeeded: 50000,
+              gasUsed: 49800,
+              gasPrice: '1',
+            },
+          ],
+        })
+      )
+      jest.mocked(getFeatureGate).mockReturnValue(true)
```

Comment:
specify the feature gate here. E.g., `jest.mocked(getFeatureGate).mockImplementation(gate => gate === <gate-name>)`, so this can be cleaned up when removing a gate. Also should this be reset in a beforeEach so it doesn't affect other tests?

Reply:
already had it reset in the beforeEach, but I'll do the mockImplementation in both